PR for SRVKS-1180: Update the information about `secret filtering` for net-istio and net-kourier (#71721)
openshift/openshift-docs: modules/serverless-ossm-secret-filtering-net-istio.adoc

:_content-type: PROCEDURE
[id="serverless-ossm-secret-filtering-net-istio_{context}"]
= Improving net-istio and net-kourier memory usage by using secret filtering for {SMProductShortName}

> By default, the link:https://aly.arriqaaq.com/kubernetes-informers/[informers] implementation for the Kubernetes `client-go` library fetches all resources of a particular type. This can lead to a substantial overhead when many resources are available, which can cause the Knative `net-istio` and `net-kourier` ingress controllers to fail on large clusters due to memory leaking. However, a filtering mechanism is available for the Knative `net-istio` and `net-kourier` ingress controllers, which enable the controllers to only fetch Knative related secrets. You can enable this mechanism by adding an annotation to the `KnativeServing` custom resource (CR).

Reviewer: ```suggestion
By default, the link:https://aly.arriqaaq.com/kubernetes-informers/[informers] implementation for the Kubernetes `client-go` library fetches all resources of a particular type. This can lead to a substantial overhead when many resources are available, which can cause the Knative `net-istio` and `net-kourier` ingress controllers to fail on large clusters due to memory leaking. However, a filtering mechanism is available for the Knative `net-istio` and `net-kourier` ingress controllers, which enable the controllers to only fetch Knative related secrets.
```

Reviewer: The point is that now this mechanism is enabled by default.

Author: True that! I have removed it.

> The secret filtering is enabled by default on the {ServerlessOperatorName} side. An environment variable, `ENABLE_SECRET_INFORMER_FILTERING_BY_CERT_UID=true`, is `true` by default to the `net-istio` and `net-kourier` controller pods.

Reviewer: ```suggestion
The secret filtering is enabled by default on the {ServerlessOperatorName} side. An environment variable, `ENABLE_SECRET_INFORMER_FILTERING_BY_CERT_UID=true`, is added by default to the `net-istio` and `net-kourier` controller pods.
```

Reviewer: Is this what was supposed to be there? Having `true` in that place doesn't make sense to me.

Author: Updated!

* Install the {ServerlessOperatorName} and Knative Serving.
* Install the OpenShift CLI (`oc`).

> You can disable the secret filtering by setting `ENABLE_SECRET_INFORMER_FILTERING_BY_CERT_UID=false` via `workloads` in the `KnativeServing` custom resource (CR).

Reviewer 2: ```suggestion
You can disable the secret filtering by setting the `ENABLE_SECRET_INFORMER_FILTERING_BY_CERT_UID` variable to `false` in the `KnativeServing` custom resource (CR).
```

Reviewer 2: I think specifying the name of the variable (`ENABLE_SECRET_INFORMER_FILTERING_BY_CERT_UID`) and CR would suffice here, since the example configuration is shown below.

        sidecar.istio.io/rewriteAppHTTPProbers: "true"
      name: autoscaler
----

> 

Author: Thank you so much I will try to merge both in one section. The example which we have used of KnativeServing CR for `net-istio`

![image](https://github.com/openshift/openshift-docs/assets/43639538/941d63b5-2c06-4057-aa30-610b519ad63d)

can you provide the same for `net-kourier`? So that we will be having one example each for both and I will include in the doc?

        sidecar.istio.io/rewriteAppHTTPProbers: "true"
      name: autoscaler
----

> You can disable the secret filtering by setting `ENABLE_SECRET_INFORMER_FILTERING_BY_CERT_UID=false` via `workloads`in the `KnativeServing` custom resource (CR).

Reviewer: ```suggestion
You can disable the secret filtering by setting `ENABLE_SECRET_INFORMER_FILTERING_BY_CERT_UID=false` via `workloads` in the `KnativeServing` custom resource (CR).
```

Author: Thank you fo capturing this typo. Updated!
